src/store/actions/addMovie.ts

```diff
@@ -0,0 +1,39 @@
+import axios from "axios";
+import { AppDispatch } from "../store";
+import { IMovie } from "../../interfaces/IMovie";
+import {
+  ADD_MOVIE_ERROR,
+  ADD_MOVIE_STARTED,
+  ADD_MOVIE_SUCCESS,
+  AddMovieActionType,
+  ADD_MOVIE_FINISHED,
+} from "../types/addMovie";
+
+const API_URL = "http://localhost:4000";
+
+export const addMovie = (movieData: IMovie) => (dispatch: AppDispatch) => {
+  dispatch(addMovieStarted());
+
+  axios
```

Comment:
it's a good practice to create service to deal with CRUD operations.
something like it:
api.addMovie(data);
api.getAllMovies();
and so on

It should encapsulate axios logic and reduce duplication of code. (API_URL for example).